cezo_fl/aggregation.py

```diff
@@ -0,0 +1,34 @@
+import torch
+
+
+def mean(num_sample_clients: int, local_grad_scalar_list):
+    grad_scalar: list[torch.Tensor] = []
+    for each_client_update in zip(*local_grad_scalar_list):
+        grad_scalar.append(sum(each_client_update).div_(num_sample_clients))
+    return grad_scalar
+
+
+def median(local_grad_scalar_list):
+    grad_scalar: list[torch.Tensor] = []
+    for each_client_update in zip(*local_grad_scalar_list):
+        each_client_tensor = torch.stack(each_client_update)
+        median = torch.median(each_client_tensor, dim=0).values
+        grad_scalar.append(median)
+    return grad_scalar
+
+
+# delete the biggest f and smallest f elements,
+# and then calculate the average of the rest of elements
+def trim(num_sample_clients: int, local_grad_scalar_list, f=1):
+    grad_scalar: list[torch.Tensor] = []
+    for each_client_update in zip(*local_grad_scalar_list):
+        each_client_tensor = torch.stack(each_client_update)
+        sorted_tensor, _ = torch.sort(each_client_tensor, dim=0)
+        trimmed_tensor = sorted_tensor[f : num_sample_clients - f]
+        trimmed_mean = trimmed_tensor.mean(dim=0)
```

Comment:
i see the point. Looks correct to be. But to give my two cents. maybe we prefer to eliminate the biggest grad in terms of magnitude.

For example, suppose we have `[0.1, 1.2, 3.2, 1000, 1001]`, we want to remove "1000, 1001" instead of "0.1" and "1001", right?

Reply:
This trim function cannot make sure that each time it is optimal.